src/nccl_ofi_net.c

```diff
@@ -2745,6 +2753,22 @@ static ncclResult_t ofi_iflush(void* recvComm, void* buffer, int size,
 	if (ofi_nccl_gdr_flush_disable() || !support_gdr)
 		goto exit;
 
+#if CUDART_VERSION >= 11030
+	if (ofi_nccl_cuda_flush_enable()) {
+		cudaError_t cuda_ret = cudaDeviceFlushGPUDirectRDMAWrites(
```

Comment:
Is `cudaDeviceFlushGPUDirectRDMAWrites` safe for systems where there are multiple PCIe switches between NIC and GPU? I understand that NCCL doesn't use GPUDirect RDMA for such cases but the users can configure it such using `NCCL_NET_GDR_LEVEL`.

I suspect that with multiple PCIe switches, NIC's completion to the host proxy thread might not guarantee that datais delivered to the GPU.

Reply:
I'm not enough of a PCIe expert to say whether the case you mentioned loses PCIe ordering between data buffer writes and the completion event. This is strictly a GPU memory fence and it assumes data was already delivered to the GPU. I can update the comment in the code and README to reflect this. Would that cover it?

Reply:
there are both cases where it is safe and cases where it is not safe.  The GPU must be directly connected to the switch where the path splits between host and GPU (there can be as many switches after that branch on the host path as desired).  It's probably not worth codifying this exception and easiest to say it's only safe to use the cudaDeviceFlush if NIC and GPU are directly connected to the same PCIe switch.